packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/hooks/FormDescriptionEditorContext.tsx

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import React from 'react';
+import { GQLFormDescriptionEditor } from '../FormDescriptionEditorEventFragment.types';
+import { FormDescriptionEditorContextValue } from './useFormDescriptionEditor.types';
+
+export const FormDescriptionEditorContext = React.createContext<FormDescriptionEditorContextValue>(null!);
```

Comment:
React.createContext *needs* a value, but at the time this is defined there is no meaningful value `FormDescriptionEditorContextValue` that I can give it.
The only options I see are:
* use `React.createContext<FormDescriptionEditorContextValue | undefined>` but then all users have to test for `undefined`
* set a non-null default `FormDescriptionEditorContextValue` value which is actually completely meaningless but type-checks:
```js
const defaultValue: FormDescriptionEditorContextValue = {
  editingContextId: '',
  representationId: '',
  readOnly: false,
  formDescriptionEditor: { id: '', pages: [] },
};
```
* use this "trick" to bypass the null-check just for this location.

I can revert to the second option if you prefer, it's what we have in e.g. `DiagramContext` or `DeckContext`. It just felt uglier in this can because of the additional level of "broken but makes TS happy" for `formDescriptionEditor`.